apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/EntrySpan.java

```diff
@@ -96,7 +96,7 @@ public AbstractTracingSpan setComponent(String componentName) {
 
     @Override
     public AbstractTracingSpan setOperationName(String operationName) {
-        if (stackDepth == currentMaxDepth) {
+        if (stackDepth == currentMaxDepth || isInAsyncMode) {
```

Comment:
I think you should add this to ExitSpan?

Reply:
ok, If exitspan also supports this scenario

Reply:
I think both of them are similar cases. Just for different plugins.